### PR TITLE
Implement react router

### PR DESCRIPTION
**Pull request by ZinaienkovaHanna**
Added react-router

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import './Button.css';
+
+interface ButtonProps {
+    path: string;
+    size: number;
+    className?: string;
+    onClick?: () => void;
+}
+
+const Button: FC<ButtonProps> = ({ path, size, className, onClick }) => {
```
Comment (MakkOliinyk) [inline src/components/Button/Button.tsx]: IconButton

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { FC } from 'react';
+import Icon from '@mdi/react';
+
+import './Button.css';
+
+interface ButtonProps {
+    path: string;
```
Comment (MakkOliinyk) [inline src/components/Button/Button.tsx]: iconPath

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    size: number;
```
Comment (MakkOliinyk) [inline src/components/Button/Button.tsx]: iconSize

File: src/components/ButtonBar/ButtonBar.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    mdiBookmarkOutline,
+    mdiYoutube,
+} from '@mdi/js';
+import Button from '../Button';
+
+interface ButtonBarProps {
+    movieId: number;
+}
+
+const ButtonBar: FC<ButtonBarProps> = ({ movieId }) => {
```
Comment (MakkOliinyk) [inline src/components/ButtonBar/ButtonBar.tsx]: MovieActions

File: src/components/IconDot/IconDot.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { FC } from 'react';
+import Icon from '@mdi/react';
+import { mdiCircleSmall } from '@mdi/js';
+
+import './IconDot.css';
+
+const IconDot: FC = () => {
```
Comment (MakkOliinyk) [inline src/components/IconDot/IconDot.tsx]: DotDivider

File: src/components/Label/Label.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import Icon from '@mdi/react';
+import Paragraph from '../Paragraph';
+
+import './Label.css';
+
+interface LabelProps {
+    path: string;
+    size: number;
+    text: string | number;
+    className?: string;
```
Comment (MakkOliinyk) [inline src/components/Label/Label.tsx]: textClassName

File: src/components/Paragraph/Paragraph.tsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+import './Paragraph.css';
+
+interface ParagraphProps {
+    text: string | number;
+    className?: string;
+}
+
+const Paragraph: FC<ParagraphProps> = ({ text, className }) => {
+    return <p className={`paragraph ${className}`}>{text}</p>;
```
Comment (MakkOliinyk) [inline src/components/Paragraph/Paragraph.tsx]: add title attribute

File: src/components/MovieInfo/MovieInfo.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import './MovieInfo.css';
+
+interface MovieInfoProps {
+    releaseDate: string;
+    genres: { id: number; name: string }[];
+    runtime: number;
+}
+
+const MovieInfo: FC<MovieInfoProps> = ({ releaseDate, genres, runtime }) => {
+    const genresInfo = genres.map((genre) => genre.name).join(', ');
```
Comment (MakkOliinyk) [inline src/components/MovieInfo/MovieInfo.tsx]: move to utils

File: src/components/MoviePage/MoviePage.tsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+                    />
+                    <Label
+                        path={mdiAccountVoice}
+                        size={1}
+                        text={movie.voteCount}
+                        className={'padding_large'}
+                    />
+                    <ButtonBar movieId={movie.id} />
+                </div>
+                <Paragraph text={movie.tagline} />
```
Comment (MakkOliinyk) [inline src/components/MoviePage/MoviePage.tsx]: Paragraph -> P

Comment (MakkOliinyk) [reply src/components/MoviePage/MoviePage.tsx]: text -> children

File: src/pages/Movie.tsx [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import { FC } from 'react';
+import { useLoaderData, Params } from 'react-router-dom';
+import MoviePage from '../components/MoviePage';
+import { getMovieById } from '../services';
+import { MovieType } from '../types/moviesTypes';
+
+export const loader = async ({ params }: { params: Params<'movieId'> }) => {
```
Comment (MakkOliinyk) [inline src/pages/Movie.tsx]: Move to a separate file

**Comment by ZinaienkovaHanna** [comment]
added react-router

File: src/components/Slider/Slider.tsx [hunk 11]
```diff
@@ -16,47 +8,32 @@ interface SliderProps {
...
-    const [cards, setCards] = useState<React.ReactNode[]>([]);
-    const [offset, setOffset] = useState(0);
+    const CARDS_WIDTH = 1200;
+    const MAX_SLIDES = 5;
+    const [pxOffset, setPxOffset] = useState(0);
+    const childArray = React.Children.toArray(children);
 
     const goToPrevSlides = () => {
-        setOffset((currentOffset) => {
+        setPxOffset((currentOffset) => {
```
Comment (MakkOliinyk) [inline src/components/Slider/Slider.tsx:18]:
```
const calculatedOffset = pxOffset + CARDS_WIDTH;
const newOffset = Math.min(calculatedOffset, 0);

setPxOffset(newOffset);
```

File: src/pages/Home.tsx [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+                        className={
+                            showToday
+                                ? 'button_today active_button'
+                                : 'button_today'
+                        }
+                    >
+                        today
+                    </button>
+                    <button
+                        onClick={handleWeekButtonClick}
```
Comment (MakkOliinyk) [inline src/pages/Home.tsx:42]: () => setShowToday(false)

File: src/pages/loader.tsx [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+import {
+    getTrendingMovies,
+    getMovies,
+    getMovieById,
+    getTvSeries,
+    getMovieTrailerById,
+} from '../services';
+
+export const homeLoader = async () => {
+    const moviesDay = await getTrendingMovies('day');
```
Comment (MakkOliinyk) [inline src/pages/loader.tsx:11]: todayMovies

**Pull request merged**